### ISSUE TITLE
DNS zone import creates absolute names instead of relative

### ISSUE THREAD

**Issue by rocketraman** [Azure/azure-cli#2752]
### Environment summary

**Install Method:** `curl -L https://aka.ms/InstallAzureCli | bash`

**CLI Version:** 
```
azure-cli (2.0.2)

acr (2.0.0)
acs (2.0.2)
appservice (0.1.2)
batch (2.0.0)
cloud (2.0.0)
component (2.0.0)
configure (2.0.2)
container (0.1.2)
core (2.0.2)
documentdb (0.1.2)
feedback (2.0.0)
find (0.0.1b1)
iot (0.1.2)
keyvault (2.0.0)
lab (0.0.1)
monitor (0.0.1)
network (2.0.2)
nspkg (2.0.0)
profile (2.0.2)
redis (0.1.1b3)
resource (2.0.2)
role (2.0.1)
sql (2.0.0)
storage (2.0.2)
vm (2.0.2)

Python (Linux) 2.7.13 (default, Jan 13 2017, 10:15:16) 
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

**OS Version:** Linux Fedora 24 (4.9.13-101.fc24.x86_64)

**Shell Type:** bash

### Description

When running a zone import e.g.:

```
az network dns zone import -g my-rg -n mydomain.com -f mydomain-zonefile.txt
```

where `mydomain-zonefile.txt` has the following content:

```
@     600     IN      SOA     ignored.com. hostmaster.mydomain.com. (
                                2017040400
                                28800
                                7200
                                604800
                                600
                                )

test    3600    IN      A       2.2.2.2
```

The relative domain record "test" is imported into Azure DNS as absolute value `test.mydomain.com` instead of relative value `test`, meaning that lookup `test.mydomain.com.mydomain.com` works and `test.mydomain.com` fails.

**Comment by jtuliani** [comment]
Thanks for flagging, will take a look

**Comment by tjprescott** [comment]
@jtuliani I've taken a look at this. The CLI expands these entries to their fully qualified form when it parses the zone file. Then, to create the record sets, it strips the trailing dot. This is why his entry resolves to "test.mydomain.com.mydomain.com". If I try to create the record set with the trailing dot, I get a service error.

I could strip the origin back off (making them relative again), which would fix the user's original issue, but then how would you handle something like:

```
@     600     IN      SOA     ignored.com. hostmaster.mydomain.com. (
                                2017040400
                                28800
                                7200
                                604800
                                600
                                )

test    3600    IN      A       2.2.2.2
$ORIGIN somethingelse.com
other
```
Now other would not resolve to other.somethingelse.com.  Thoughts?

**Comment by jtuliani** [comment]
Hi Travis. The logic should be something like:

1. Parse the record name
2. If the record name is relative, add on the current value of $ORIGIN (which can change from time to time), to get the FQDN
3. Remove the zone name (which stays constant throughout the import)
4. Remove trailing dot
5. Use the result as the record name (or if you have an empty name at this point, use '@' instead).

At step 3, the FQDN should *always* end with the zone name. If not, then there was an error earlier, either in the record name or in the value of $ORIGIN.

**Comment by tjprescott** [comment]
Yes, but what happens in step 2 when the record name is not relative?

**Comment by jtuliani** [comment]
If the record name is not relative, then it IS the FQDN so step 2 just uses it as-is. The value of $ORIGIN is not used in this case. The FQDN must end with the zone name, otherwise an error will occur at Step 3.

**Comment by tjprescott** [comment]
So if you have a full qualified name that doesn't end with the zone name, that's an error?

**Comment by rocketraman** [comment]
> So if you have a full qualified name that doesn't end with the zone name, that's an error?

@tjprescott The CLI is explicitly importing into that zone. So if the import file contains unrelated zone information, I would say you could just ignore it, with a warning to the user that it was skipped due to a zone mismatch.

**Comment by tjprescott** [comment]
@rocketraman that's what it currently does after I merged #2825.

**Comment by jtuliani** [comment]
Yup, warn+skip is OK
